caikit/runtime/service_generation/primitives.py

```diff
@@ -82,6 +82,36 @@ def to_primitive_signature(
     return primitives
 
 
+def to_output_dm_type(arg_type: type) -> str:
+    """Helper function that determines the right data model type to use based on
+    the python type of the argument.
+    """
+
+    # Decompose this type using typing to determine if it's a useful typing hint
+    typing_origin = get_origin(arg_type)
+    typing_args = get_args(arg_type)
+
+    # If this is a data model type, no need to do anything
+    if isinstance(arg_type, type) and issubclass(arg_type, DataBase):
+        return arg_type
+
+    # Handle Unions by looking for a data model object in the union
+    if typing_origin is Union:
+        dm_types = [
+            arg
+            for arg in typing_args
+            if inspect.isclass(arg) and issubclass(arg, DataBase)
+        ]
+        if dm_types:
+            log.debug2(
+                "Found data model types in Union: [%s], taking first one", dm_types
+            )
+            return to_output_dm_type(dm_types[0])
```

Comment:
Hmm. Yeah I thought about that, I left it in case we want to expand on this later on with more types of arguments. I can make it easy if that makes better sense